reaction_network/fjss.py

```diff
@@ -688,83 +690,95 @@ def build_model_gurobi(self):
 
             # not all the operations will need the work shift constraints
             if self.operations_subset_indices:
-                operations_subset = {i: self.operations[i] for i in self.operations_subset_indices}
-                # n_opt_subset = len(operations_subset)
+                # take the subset of the original operations ordered dictionary based on the
+                # operations_subset_indices
+                # the subset ordered dictionary follow the same order as the original ordered
+                # dictionary
+                operations_subset = [(list(self.operations.keys())[i], list(self.operations.values())[i])
+                                     for i in self.operations_subset_indices]
+                operations_subset = OrderedDict(operations_subset)
+
             else:
                 # n_opt_subset = n_opt
                 operations_subset = self.operations
 
             eps = 1e-5
             value_m = self.horizon*10
```

Comment:
I thought there was an equation for estimate M, why are you using the 10*horizon here?